### PR TITLE
Add linting for HTML files

### LINKED ISSUE #10: Check/fix HTML files

### PR DESCRIPTION
**Pull request by eheikes**
fixes #10

Uses [HTMLHint](https://github.com/yaniswang/HTMLHint) for the basic checking, plus adds a few custom rules. See the `config/htmlhint.hjson` file for details of what is checked.

cc @Banno/ux

## How to Test

* `npm run lint` and `npm test` should pass.
* You can run it against our projects to see the output:
  * `node cli.js /path/to/banno-web/src/**/*.html`
  * `node cli.js /path/to/platform-ux/src/index.html /path/to/platform-ux/src/*/*.html`

## Notes

It's not possible to whitelist tags the way you can for attributes. So we can whitelist SVG attributes like `viewBox` but not SVG tags like `clipPath`. This has only been used in 1 file so far though.

## Discussion Points

* I set `space-tab-mixed-disabled` to tabs, to be consistent with our JS guidelines. But we don't really use tabs for HTML files -- would it be better to be inconsistent here and potentially more confusing?
* Is the HTMLHint library good enough? It looks like not much development is going on with it, but it works pretty well currently, and for any changes we should be able to fork and modify it w/o too much trouble.

### CODE REVIEW

**Comment by jrunning** [comment]
> I set `space-tab-mixed-disabled` to tabs, to be consistent with our JS guidelines. But we don't really use tabs for HTML files -- would it be better to be inconsistent here and potentially more confusing?

banno-web [uses spaces everywhere](https://github.com/Banno/banno-web/blob/master/src/js/bootstrap.js), so we're inconsistent no matter what we do.

> Is the HTMLHint library good enough? It looks like not much development is going on with it, but it works pretty well currently, and for any changes we should be able to fork and modify it w/o too much trouble.

FWIW the venerable Tidy got updated to handle custom elements [just a few weeks ago](https://github.com/htacg/tidy-html5/issues/119), but it hasn't been released yet. It looks like HTMLHint is a lot easier to add custom rules to, though.

File: linters/htmlhint.js [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+	description: '<html> tags must have a "lang" attribute.',
+	init: configureDocLang
+});
+htmlhint.addRule({
+	id: 'link-href',
+	description: 'Links must have an href and it must be valid.',
+	init: configureLinkHref
+});
+htmlhint.addRule({
+	id: 'meta-charset',
```
Comment (jrunning) [inline linters/htmlhint.js]: Nitpick: This should probably be named `meta-charset-utf8`.

File: linters/htmlhint.js [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+const config = parseJson(__dirname + '/../config/htmlhint.hjson');
+
+// Add our custom rules.
+htmlhint.addRule({
+	id: 'doc-lang',
+	description: '<html> tags must have a "lang" attribute.',
+	init: configureDocLang
+});
+htmlhint.addRule({
+	id: 'link-href',
```
Comment (jrunning) [inline linters/htmlhint.js]: In ESLint it's conventional for rules supplied by plugins to be prefixed with the plugin name, e.g. `jsdoc/check-param-names` or `jasmine/no-focused-tests`. This makes it easy to tell at a glance where to find the documentation for a particular rule. Would it be worth doing the same here?

Comment (eheikes) [inline linters/htmlhint.js]: That's an idea. Speaking of documentation, I should probably add docs about these rules to the project.

**Comment by eheikes** [comment]
* Updated for Jordan's suggestions
* Added docs for custom rules
* Added fix so that `<a href>` is allowed
* Changed HTML indentation to spaces following https://github.com/Banno/wiki/pull/404

**Comment by jrunning** [comment]
This LGTM 👍

**Pull request merged**